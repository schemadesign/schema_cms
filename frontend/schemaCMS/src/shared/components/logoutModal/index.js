export { LogoutModal } from './logoutModal.component';
