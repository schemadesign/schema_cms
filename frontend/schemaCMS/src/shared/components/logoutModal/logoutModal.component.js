import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { modalStyles, Modal, ModalTitle, ModalActions } from '../modal/modal.styles';
import { BackButton, ConfirmLink } from '../navigation';
import messages from './logoutModal.messages';

export class LogoutModal extends PureComponent {
  static propTypes = {
    logoutModalOpen: PropTypes.bool.isRequired,
    redirectUrl: PropTypes.string.isRequired,
    onAction: PropTypes.func.isRequired,
  };

  render() {
    const { logoutModalOpen, onAction: handleAction, redirectUrl } = this.props;

    return (
      <Modal isOpen={logoutModalOpen} contentLabel="Confirm Logout" style={modalStyles}>
        <ModalTitle>
          <FormattedMessage {...messages.title} />
        </ModalTitle>
        <ModalActions>
          <BackButton onClick={handleAction}>
            <FormattedMessage {...messages.cancel} />
          </BackButton>
          <ConfirmLink to={redirectUrl} onClick={handleAction}>
            <FormattedMessage {...messages.confirm} />
          </ConfirmLink>
        </ModalActions>
      </Modal>
    );
  }
}
