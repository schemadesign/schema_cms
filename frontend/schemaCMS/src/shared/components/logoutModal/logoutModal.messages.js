/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'shared.components.logoutModal.title',
    defaultMessage: 'Are you sure you want to log out?',
  },
  cancel: {
    id: 'shared.components.logoutModal.cancel',
    defaultMessage: 'Cancel',
  },
  confirm: {
    id: 'shared.components.logoutModal.confirm',
    defaultMessage: 'Confirm',
  },
});
