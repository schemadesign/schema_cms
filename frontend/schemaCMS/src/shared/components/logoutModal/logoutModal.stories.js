import React from 'react';
import { storiesOf } from '@storybook/react';

import { LogoutModal } from './logoutModal.component';

const defaultProps = {};

storiesOf('Shared Components|LogoutModal', module).add('Default', () => <LogoutModal {...defaultProps} />);
