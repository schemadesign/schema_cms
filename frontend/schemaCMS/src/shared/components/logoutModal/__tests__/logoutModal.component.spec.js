import React from 'react';
import { shallow } from 'enzyme';

import { LogoutModal } from '../logoutModal.component';

describe('LogoutModal: Component', () => {
  const defaultProps = {
    logoutModalOpen: true,
    onConfirm: Function.prototype,
    onCancel: Function.prototype,
  };

  const component = props => <LogoutModal {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should be hidden', () => {
    const wrapper = render({
      logoutModalOpen: false,
    });
    global.expect(wrapper).toMatchSnapshot();
  });
});
