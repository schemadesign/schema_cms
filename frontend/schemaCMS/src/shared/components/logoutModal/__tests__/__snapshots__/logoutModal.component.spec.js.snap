// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutModal: Component should be hidden 1`] = `
<Styled(Modal)
  contentLabel="Confirm Logout"
  isOpen={false}
  style={
    Object {
      "content": Object {
        "WebkitOverflowScrolling": "touch",
        "alignItems": "center",
        "border": null,
        "borderRadius": "4px",
        "boxShadow": "0 4px 8px 0 rgba(0, 0, 0, 0.5)",
        "display": "flex",
        "flexDirection": "column",
        "left": "50%",
        "maxWidth": "600px",
        "outline": "none",
        "overflow": "auto",
        "position": "absolute",
        "right": "10px",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
        "width": "calc(100% - 20px)",
      },
      "overlay": Object {
        "backgroundColor": "rgba(29, 29, 32, 0.75)",
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 10000,
      },
    }
  }
>
  <Styled(H1)>
    <FormattedMessage
      defaultMessage="Are you sure you want to log out?"
      id="shared.components.logoutModal.title"
      values={Object {}}
    />
  </Styled(H1)>
  <styled.div>
    <BackButton
      disabled={false}
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Cancel"
        id="shared.components.logoutModal.cancel"
        values={Object {}}
      />
    </BackButton>
    <NextButton
      loading={false}
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Confirm"
        id="shared.components.logoutModal.confirm"
        values={Object {}}
      />
    </NextButton>
  </styled.div>
</Styled(Modal)>
`;

exports[`LogoutModal: Component should render correctly 1`] = `
<Styled(Modal)
  contentLabel="Confirm Logout"
  isOpen={true}
  style={
    Object {
      "content": Object {
        "WebkitOverflowScrolling": "touch",
        "alignItems": "center",
        "border": null,
        "borderRadius": "4px",
        "boxShadow": "0 4px 8px 0 rgba(0, 0, 0, 0.5)",
        "display": "flex",
        "flexDirection": "column",
        "left": "50%",
        "maxWidth": "600px",
        "outline": "none",
        "overflow": "auto",
        "position": "absolute",
        "right": "10px",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
        "width": "calc(100% - 20px)",
      },
      "overlay": Object {
        "backgroundColor": "rgba(29, 29, 32, 0.75)",
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 10000,
      },
    }
  }
>
  <Styled(H1)>
    <FormattedMessage
      defaultMessage="Are you sure you want to log out?"
      id="shared.components.logoutModal.title"
      values={Object {}}
    />
  </Styled(H1)>
  <styled.div>
    <BackButton
      disabled={false}
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Cancel"
        id="shared.components.logoutModal.cancel"
        values={Object {}}
      />
    </BackButton>
    <NextButton
      loading={false}
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Confirm"
        id="shared.components.logoutModal.confirm"
        values={Object {}}
      />
    </NextButton>
  </styled.div>
</Styled(Modal)>
`;
